connect_ext_ppr/validator.py

```diff
@@ -0,0 +1,64 @@
+from fastapi import status
+
+from connect_ext_ppr.errors import ExtensionValidationError
+
+
+def validate_deployment(deployment, account_id):
+    """
+    Validates that the deployment belongs to the account
+    """
+    if deployment.account_id != account_id:
+        raise ExtensionValidationError.VAL_001(
+            format_kwargs={
+                'field': 'deployment',
+                'id': deployment.id,
+            },
+            status_code=status.HTTP_400_BAD_REQUEST,
+        )
+
+
+def validate_ppr_version_belongs_to_deployment(ppr, deployment):
+    """
+    Validates that the ppr can be asociated to DR
+    """
+    if ppr.deployment != deployment.id:
+        raise ExtensionValidationError.VAL_001(
+            format_kwargs={'field': 'ppr', 'id': ppr.id},
+            status_code=status.HTTP_400_BAD_REQUEST,
+        )
+
+
+def validate_dr_marketplaces(dr_marketplaces, dep_marketplaces):
+    """
+    Validates that all the DR's marketplaces belong to the associated deployment
+    """
+    diff = list(set(dr_marketplaces) - set(dep_marketplaces))
+    if diff:
+        diff.sort()
+        raise ExtensionValidationError.VAL_002(
+            format_kwargs={
+                'field': 'marketplaces',
+                'values': diff,
+            },
+            status_code=status.HTTP_400_BAD_REQUEST,
+        )
+
+
+def validate_marketplaces_ppr(ppr, dr_marketplaces, dep_marketplaces):
```

Comment:
I think there should be a check, that provided PPR belongs to the deployment also. Otherwise for example user via API can provide PPR from another Hub or another Product even. Do we have such kind of check?